### DISCUSSION TITLE
Blade Component - Pass property with colon in the middle and access from class

### DISCUSSION THREAD

**Discussion by pkeogan** [livewire/livewire#4963] [1. Help]
not exactly and issue with livewire, but it involves it. Maybe someone here has encountered this issue?

I have the following code in a Livewire Component 

```
<x-input.color
	wire:model="foo.name"
	wire-test="foo.name"
/>
```

So I am trying to use a Blade Component and pass both properties to the class.

```
class Color extends Component
{
	public $wireTest;
	public $wireModel;
	
	public function __construct($wireTest = null, $wireModel = null)
	{
		$this->wireTest = $wireTest; // will resolve
		$this->wireModel = $wireModel; // wont resolve
		dump($this->data()); // will resolve the attribute bag that's empty
	}
	
	public function render()
	{
	         dump($this->data()); // will resolve the attribute bag that's empty
		 return view('components.input.color');
	}

}
```

```
//color.blade.php
<input {{$attributes}} ... />
//$attributes will contain 'wire:model' => 'foo.name'
```

I cannot figure out how to pass access a property with a : in the middle of the name: ie ```wire:model``` within the class

Is there anyway I can access the ```wire:model``` attribute within the component class?

**Comment by joshhanley** [comment]
@pkeogan try using an anonymous component instead of one backed by a class. Or maybe don't define it as a property and instead see if you can access the attributes bag inside the class and pull it from that.

Hope it helps!
